### DISCUSSION TITLE
Ape support

### DISCUSSION THREAD

**Discussion by m0t0rh3ad** [andresch/trackfs#28] [Ideas]
Hello, first of all, thanks a lot for this very useful product and your work!
I want to ask, is it possible to add APE files support for APE+CUE splitting?

**Comment by andresch** [comment]
Hi m0t0rh3ad,

Thanks for giving trackfs a try. 

I have no experience with APE. I quickly checked that the mutagen (the meta-data library trackfs is using) is supporting APE. What I could not find is lightweight and fast approach to extract a track from an APE album file under (alpine) linux.

I have found two ways that might work, but non of them is straight forward/ without problems:

- A combination of [shntool](http://shnutils.freeshell.org/shntool/) and [mac](https://github.com/fernandotcl/monkeys-audio), but unfortunately mac doesn't seem to be supported on Alpine-Linux, which is used as basis for the docker container. I could see if I could build modify my docker file to build mac from source under alpine, but based on my experience this is often quite tricky under alpine). Alternatively I would have to migrate from alpine to debian (which would make the trackfs container substantially larger)

- ffmpeg, but (a) from what I would find it seems to only support reading APE, so the output would then be something else and (b) ffmpgeg is a huge package and the size for trackfs would explode. You might want to see if [FFmpegfs](https://github.com/nschlia/ffmpegfs) would work for you. Cuesheet support is already on the masterbranch and might be officially released soon.

Do you know of any command line tool that I would use to extact a single track from an APE file?
